modules/javafx.graphics/src/main/java/javafx/css/StyleClass.java

```diff
@@ -27,6 +27,7 @@
 /**
  * A class that contains {@code StyleClass} information.
  * @since 9
+ * @deprecated for removal
  */
 public final class StyleClass {
```

Comment:
Needs the `@Deprecated(forRemoval = true)` annotation on the class, and the doc tag needs an explanation of why it's being removed ("erroneously exposed", "no longer needed", "flawed from the start"...) and what should be used instead, if at all.

Reply:
I added it.

Technically, we could leave this class in, but it would just be a dead class that serves no purpose within FX, at least, once the `SimpleSelector` and `CompoundSelector` classes have been moved to internal packages.

I'm wondering if I should add any deprecations in this PR at all, as now it requires a CSR.  This could be all handled in the PR that moves these classes to internal packages.

Reply:
Ideally, you would not need any deprecations in this PR.